recipes/liblsl/all/conanfile.py

```diff
@@ -47,54 +37,57 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LSL_BUILD_STATIC"] = not self.options.shared
+        tc.variables["LSL_BUNDLED_BOOST"] = False
+        tc.variables["LSL_BUNDLED_PUGIXML"] = False
+        tc.variables["lslgitrevision"] = "v" + self.version
+        tc.variables["lslgitbranch"] = "master"
+        tc.generate()
+        cd = CMakeDeps(self)
+        cd.generate()
 
     def _patch_sources(self):
         if not self.options.shared:
             # Do not force PIC
-            tools.replace_in_file(
-                os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, "CMakeLists.txt"),
                 "set(CMAKE_POSITION_INDEPENDENT_CODE ON)",
                 ""
             )
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["LSL_BUILD_STATIC"] = not self.options.shared
-        self._cmake.definitions["LSL_BUNDLED_BOOST"] = False
-        self._cmake.definitions["LSL_BUNDLED_PUGIXML"] = False
-        self._cmake.definitions["lslgitrevision"] = "v" + self.version
-        self._cmake.definitions["lslgitbranch"] = "master"
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
-
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
         cmake.install()
 
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "lslver*")
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rm(self, "lslver*", os.path.join(self.package_folder, "bin"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
 
         self._create_cmake_module_alias_targets(
             os.path.join(self.package_folder, self._module_file_rel_path),
             {"lsl": "LSL::lsl"}
```

Comment:
just to says that it doesn't make sense given package_info() content

Reply:
you should set CMake target for CMakeDeps in package_info() by the way. But since it was wrong in cmake_find_package as I said, I don't really understand what should be the correct target name.

Reply:
ok so target should be `LSL::lsl`

Reply:
Like so? caf23d63b96f9ad4f7a57efb7c31f25dcd11f08a

Reply:
nop, it should be:

`{"LSL::lsl": "lsl::lsl"}`

But components should have been used to begin with. I don't understand why reviewers of first PR didn't tell you that it was a less convoluted approach.

Reply:
I think you meant like this, right? 72cd7fedff2f40a65c5de619887d977e2e78bb5f

Reply:
I've updated my message, you can take the code snippet as is to and replace everything in package_info().
All the `_create_cmake_module_alias_targets()` stuff can be removed.

Reply:
Obviously test package should be updated accordingly with find_package(LSL REQUIRED CONFIG) and target_link_libraries(${PROJECT_NAME} PRIVATE LSL::lsl)

Reply:
Thanks a lot, I updated the code with your suggestion.